core/src/worker/heartbeat.rs

```diff
@@ -1,50 +1,128 @@
+use crate::WorkerClient;
+use crate::abstractions::dbg_panic;
 use gethostname::gethostname;
+use parking_lot::Mutex;
 use prost_types::Duration as PbDuration;
+use std::sync::{Arc, OnceLock};
 use std::time::{Duration, SystemTime};
 use temporal_sdk_core_api::worker::WorkerConfig;
 use temporal_sdk_core_protos::temporal::api::worker::v1::{WorkerHeartbeat, WorkerHostInfo};
 use tokio::sync::watch;
+use tokio::task::JoinHandle;
+use tokio::time::MissedTickBehavior;
 use uuid::Uuid;
 
+pub(crate) struct WorkerHeartbeatManager {
+    heartbeat_handle: JoinHandle<()>,
+}
+
+impl WorkerHeartbeatManager {
+    pub(crate) fn new(
+        config: WorkerConfig,
+        identity: String,
+        heartbeat_fn: Arc<OnceLock<Box<dyn Fn() -> Option<WorkerHeartbeat> + Send + Sync>>>,
+        client: Arc<dyn WorkerClient>,
+    ) -> Self {
+        let sdk_name_and_ver = client.sdk_name_and_version();
+        let (reset_tx, reset_rx) = watch::channel(());
```

Comment:
I'm realizing this should probably just be a https://docs.rs/tokio/latest/tokio/sync/struct.Notify.html